components/datetime/Cargo.toml

```diff
@@ -57,6 +57,7 @@ bench = false  # This option is required for Benchmark CI
 default = ["provider_serde"]
 bench = []
 provider_serde = ["serde", "litemap/serde"]
+provider_transform_utils = []
```

Comment:
Nit 1: Can we think of a better name?  Ideas:

- `provider_transform` (generic)
- `transform` (generic)
- `pattern_transformations` (DateTime-specific)

Nit 2: If you choose a sufficiently generic feature name, this feature may be useful in other crates as well, please document it in */components/icu/src/lib.rs* (and re-generate */components/icu/README.md*).

Reply:
Nit 1: I was thinking the name was specific towards "internals required by provider transform", and that it would be useful across the project. I'd prefer something a bit more verbose, such as `provider_transform_internals`. The idea behind this is that it's useful to expose something to the provider transformer code, but not to the public API. I'd like to make it scary to use for our end users, and "internals" suggests it shouldn't be touched.

Nit 2: This is already done:

```
//! - `provider_transform_internals`: This is code that is useful for transforming CLDR data, but
//!   may be risky to ship as a public API. For instance in DateTimeFormat code we may want to
//!   manipulate patterns, but not allow end users to do the same.
```